lib/logthrdest/logthrdestdrv.c

```diff
@@ -1120,6 +1120,18 @@ _register_driver_stats(LogThreadedDestDriver *self, StatsClusterKeyBuilder *driv
 
   gint level = log_pipe_is_internal(&self->super.super.super) ? STATS_LEVEL3 : STATS_LEVEL0;
 
+  _init_driver_sck_builder(self, driver_sck_builder);
+
+  stats_cluster_key_builder_set_name(driver_sck_builder, "output_message_delay_sample_age_seconds");
+  stats_cluster_key_builder_set_unit(driver_sck_builder, SCU_SECONDS);
+  stats_cluster_key_builder_set_frame_of_reference(driver_sck_builder, SCFOR_RELATIVE_TO_TIME_OF_QUERY);
+  self->metrics.message_delay_sample_age_key = stats_cluster_key_builder_build_single(driver_sck_builder);
```

Comment:
shouldn't this be a per-worker metric? if we have the worker=%d label in it, we can still aggregate at the prometheus level (e.g. by finding the maximum value)

Reply:
Sure, we can make it so, but I don't see the case where we would need those separately. It would be easier to implement them there, though, as the key builder is in a more suitable state there, which is a plus for me. :)

Reply:
I have moved this to the workers. Unfortunately because of a lifecycle management limitation, we will not have these counters for those threaded dest driver implementations which only support the compat worker.

Reply:
I think if a use-case comes up for this metric, we can always convert those to not use the compat layer, so I think it is fine now.